Fix hook dependencies and violating

cservenka opened the pull request:
- A number of `useEffect` hooks were missing dependency arrays
- A number of `useMemo` hooks had too many dependencies listed
- The custom `useMungeData` hook was violating the [top-level rule](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) of hooks by conditionally executing the latter two `useMemo` calls only if `stepDetails` was passed to the hook
- Replaced a few `lodash.get`s with optional chains instead

bjenkins24 commented:
@cservenka I'm not sure if this was caused by this PR or not: https://share.squarespace.com/GGu2Dqol

cservenka commented:
> @cservenka I'm not sure if this was caused by this PR or not: https://share.squarespace.com/GGu2Dqol

Interesting! I'll investigate a little closer

cservenka commented:
> @cservenka I'm not sure if this was caused by this PR or not: https://share.squarespace.com/GGu2Dqol

Turns out `useRef` saves it's value across the entire component's lifetime, even if the reference is updated. Changed the refs to be callbacks, like detailed in this example: https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node

Updated in https://github.com/AcuityScheduling/acuity-calendar/pull/57/commits/21dacf987731b4d90e649009bbe3996548ca07b5

1-2-3-4 commented on .eslintrc:
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "react-app",
+  "extends": ["react-app", "plugin:react-hooks/recommended"],
```
👍

1-2-3-4 commented on src/Calendar/components/DayGrid/utils/useTotalEventsToShow.js:
```diff
@@ -16,46 +12,49 @@ const useTotalEventsToShow = () => {
-      }
+  /**
+   * Using callbacks instead of refs as detailed here:
+   * https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
```
thx for including this resource, very helpful 👍

cesutherland replied:
:100:

1-2-3-4 reviewed (approved):
lgtm!
